In [1]:
import solution
import glob

In [2]:
count = 0

for file_name in glob.glob("./PublicData3/*.dat"):
    # if count in [6, 8]:
    #     count += 1
    #     continue
    print(f"\n\nExercise {count}:")
    autoScheduler = solution.FleetProblem() # Initialize the problem
    with open(file_name, "r") as f: 
        autoScheduler.load(f) # Load the file
        
    # print(autoScheduler)
    
    sol = autoScheduler.solve(display_solutions=True) # Solving the problem
    # print(f"Our Solution: {sol}")
    # print(f"Goal Estimates: {f_to_sol}")
    # print(f"Cost: {autoScheduler.cost(sol)}")
    
    count += 1
    
    # if count == 6:
    #     break



Exercise 0:
2 paths have been expanded and 0 paths remain in the frontier
Cost: 10.0
Goal Estimates: [10.0, 10.0]
Solution: [('Pickup', 0, 0, 20.0), ('Dropoff', 0, 0, 60.0)]


Exercise 1:
14 paths have been expanded and 13 paths remain in the frontier
Cost: 69.0
Goal Estimates: [149.0, 139.0, 139.0, 69.0, 69.0, 69.0]
Solution: [('Pickup', 0, 1, 20.0), ('Pickup', 1, 0, 20.0), ('Dropoff', 1, 0, 70.0), ('Pickup', 1, 2, 70.0), ('Dropoff', 0, 1, 80.0), ('Dropoff', 1, 2, 140.0)]


Exercise 2:
54 paths have been expanded and 53 paths remain in the frontier
Cost: 139.0
Goal Estimates: [199.0, 149.0, 99.0, 99.0, 99.0, 99.0, 99.0, 139.0]
Solution: [('Pickup', 0, 2, 30.0), ('Pickup', 0, 3, 30.0), ('Pickup', 1, 0, 20.0), ('Dropoff', 0, 3, 80.0), ('Pickup', 0, 1, 80.0), ('Dropoff', 1, 0, 70.0), ('Dropoff', 0, 1, 140.0), ('Dropoff', 0, 2, 140.0)]


Exercise 3:
134 paths have been expanded and 195 paths remain in the frontier
Cost: 144.0
Goal Estimates: [244.0, 194.0, 144.0, 104.0, 99.0, 99.0, 99.0

Exercise 0:
2 paths have been expanded and 0 paths remain in the frontier
Cost: 10.0


Exercise 1:
16 paths have been expanded and 14 paths remain in the frontier
Cost: 69.0


Exercise 2:
105 paths have been expanded and 203 paths remain in the frontier
Cost: 139.0


Exercise 3:
140 paths have been expanded and 428 paths remain in the frontier
Cost: 144.0


Exercise 4:
177 paths have been expanded and 920 paths remain in the frontier
Cost: 87.0


Exercise 5:
51 paths have been expanded and 205 paths remain in the frontier
Cost: 2.832334879439538


Exercise 7:
225 paths have been expanded and 367 paths remain in the frontier
Cost: 4.542734978545324


Exercise 9:
800 paths have been expanded and 1463 paths remain in the frontier
Cost: 139.0492583219812

In [3]:
# import solution
# solver = solution.FleetProblem()

# with open("PublicData3/ex1.dat", "r") as f:
#     solver.load(f)

# print(solver.initial.vehicles)    
# solver.initial.vehicles[0].time = 2
# solver.initial.vehicles[1].time = 1

# solver.initial.vehicles = sorted(solver.initial.vehicles)
# print(solver.initial.vehicles)

In [4]:
import numpy as np

np.ones(3)

array([1., 1., 1.])